src/lib.rs

```diff
@@ -16,61 +18,42 @@ pub extern "C" fn wapc_init() {
 
 fn validate(payload: &[u8]) -> CallResult {
     let validation_req = ValidationRequest::<Settings>::new(payload)?;
-
-    let query = "spec.containers[*].securityContext.allowPrivilegeEscalation";
-    if has_container_with_allow_privilege_escalation(query, &validation_req)? {
-        return reject_request(
-            Some(format!(
-                "User '{}' cannot create containers with allowPrivilegeEscalation enabled",
-                validation_req.request.user_info.username,
-            )),
-            None,
-        );
-    }
-
-    let query = "spec.initContainers[*].securityContext.allowPrivilegeEscalation";
-    if has_container_with_allow_privilege_escalation(query, &validation_req)? {
-        return reject_request(
-            Some(format!(
-                "User '{}' cannot create initContainers with allowPrivilegeEscalation enabled",
-                validation_req.request.user_info.username,
-            )),
-            None,
-        );
+    let pod = serde_json::from_value::<apicore::Pod>(validation_req.request.object)?;
+
+    let any_privileged_container = pod.spec
+        .map(|spec| {
+            has_privileged_container(spec.init_containers) ||
+                has_privileged_container(Some(spec.containers))
+        })
+        .unwrap_or(false);
+
+    if any_privileged_container {
+        reject_request(Some(format!(
+            "User '{}' cannot create containers with allowPrivilegeEscalation enabled",
+            validation_req.request.user_info.username,
+        )), None)
+    } else {
+        accept_request()
     }
-
-    accept_request()
 }
 
-fn has_container_with_allow_privilege_escalation(
-    query: &str,
-    validation_req: &ValidationRequest<Settings>,
-) -> anyhow::Result<bool> {
-    let containers_query = jmespatch::compile(query)
-        .map_err(|e| anyhow!("Cannot parse jmespath expression: {:?}", e,))?;
-
-    let raw_search_result = validation_req
-        .search(containers_query)
-        .map_err(|e| anyhow!("Error while searching request: {:?}", e,))?;
-    if raw_search_result.is_null() {
-        return Ok(false);
-    }
-
-    let search_result = raw_search_result.as_array().ok_or_else(|| {
-        anyhow!(
-            "Expected search matches to be an Array, got {:?} instead",
-            raw_search_result
-        )
-    })?;
-
-    Ok(search_result.contains(&Rc::new(jmespatch::Variable::Bool(true))))
+fn has_privileged_container(containers: Option<Vec<apicore::Container>>) -> bool {
+    containers
+        .unwrap_or_default()
+        .into_iter()
+        .any(|container| {
+            container.security_context
+                .map_or(false, |security_context| {
+                    security_context.allow_privilege_escalation.unwrap_or(false)
```

Comment:
I "copied" this logic as is. There's also the `privileged` attribute on the security context. Are we ignoring it in this policy in purpose? cc/ @flavio

Reply:
That's intended, we have another policy looking after the creation of privileged containers